src/app/components/ParticipatedContests/index.js

```diff
@@ -1,116 +1,95 @@
 
-function ParticipatedContests({ token, username }) {
-  const [page, setPage] = useState(1);
-  const [pastContests, setPastContests] = useState([]);
-  useEffect(() => {
-    get(
-      /*
-      `https://test.api.freecontest.net/api/v1/participations?username=${username}&offset=${
-        (page - 1) * pastContestsPerPage
-      }&limit=${pastContestsPerPage}`, */
-      'https://my-json-server.typicode.com/upi05/mock_API_for_Fc5y/participations',
-      {
-        headers: {
-          Authorization: `Bearer ${token}`,
-        },
-      },
-      true,
-    ).then((res) => {
-      // Don't forget to handle errors
-      numberOfContests = res.data.data.total;
-      setPastContests(res.data.data.participations);
+// Contants
+import { TB_CONFIG } from './config';
+import { ROUTE_CONTEST } from 'src/app/routes/constants';
+
+// Styles
+const ContestTitle = styled.h1`
+  font-weight: 700;
+  font-size: 16px;
+  color: var(--primary-default);
+  cursor: pointer;
+`;
+
+let totalContests = 0;
+
+function ParticipatedContests({ username }) {
+  const [tableConfig, setTableConfig] = useState(TB_CONFIG);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [currentPageSize, setCurrentPageSize] = useState(10);
+  const history = useHistory();
+
+  // Helper function to format table data
+  const formatTableData = (data) => {
+    return data.map((d, i) => {
+      return {
+        index: i + 1 + (currentPage - 1) * currentPageSize,
+        contestName: (
+          <ContestTitle
+            onClick={() => history.push(makeUrl(ROUTE_CONTEST, { contestName: d.contest_name }))}
+          >
+            {d.contest_title || ''}
+          </ContestTitle>
+        ),
+        ranking: `${d.rank_in_contest}/${d.contest_total_participations}`,
+        rating: (
+          <>
+            {d.rating}
+            <span
+              className={parseInt(d.rating_change, 10) < 0 ? styles.ratingRed : styles.ratingGreen}
+            >
+              ({d.rating_change})
+            </span>
+          </>
+        ),
+      };
     });
-  }, []);
+  };
 
-  const trList = pastContests.map((contest) => (
-    <tr className={styles.rowContest} key={contest.id}>
-      <th className={styles.tableNumber}>{contest.id}</th>
-      <th className={styles.tableContest}>
-        <Link to={`/contest/${contest.id}`}>{contest.contest_title}</Link>
-      </th>
-      <th className={styles.tableRanking}>
-        {contest.rank_in_contest}/{contest.contest_total_participations}
-      </th>
-      <th className={styles.tableRating}>
-        {contest.rating}
-        <span className={contest.rating_change < 0 ? styles.ratingRed : styles.ratingGreen}>
-          ({contest.rating_change})
-        </span>
-      </th>
-    </tr>
-  ));
+  useEffect(() => {
+    apiGetParticipations({
+      username,
+      offset: (currentPage - 1) * currentPageSize,
+      limit: currentPageSize,
+    }).then((res) => {
+      totalContests = res.data.total;
```

Comment:
`totalContests` should be a state, do not create a global var if it's not a constants